src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -77,8 +80,17 @@ public static void generateReposReport(List<RepoConfiguration> configs, String o
         Map<RepoLocation, List<RepoConfiguration>> repoLocationMap = groupConfigsByRepoLocation(configs);
         cloneAndAnalyzeRepos(repoLocationMap, outputPath);
 
-        Date sinceDate = cliSinceDate == null ? earliestSinceDate : cliSinceDate;
-        Date untilDate = cliUntilDate == null ? latestUntilDate : cliUntilDate;
+        Date sinceDate = (cliSinceDate.isPresent()) ? cliSinceDate.get() : null;
+        Date untilDate = (cliUntilDate.isPresent()) ? cliUntilDate.get() : new Date();
+
+        if (sinceDate == null) { // set sinceDate to one month before untilDate if sinceDate is null.
```

Comment:
This part is duplicated with the part in `RepoConfiguration.java`. Can this part be improved? i.e. Calculated the correct `since date` in `ArgsParser`?

Reply:
https://github.com/reposense/RepoSense/blob/b72388535cc1a533bbc45e6fd23bd95a13a5cc5b/src/main/java/reposense/parser/ArgsParser.java#L104-L108

Should we `.setDefault(...)` to one month before report generation date here?

If so, should we also `.setDefault(...)` for untilDate in `ArgsParser` to the date of report generation as well?

Reply:
I think it is a good idea to do so because it is a job of `ArgsParser` to get the final/real argument. The rest only need to respond to the date argument given.